vignettes/advs.Rmd

```diff
@@ -0,0 +1,392 @@
+avoid repeating information and maintaining the same content in two
+places.
+
+# Programming Workflow
+
+-   [Read in Data](#readdata)
+-   [Initial ADVS Set-up](#advs_start)
+-   [Derive Additional Parameters (e.g. `BMI` for `ADVS`)](#derive_param)
+-   [Derive Categorization Variables (`AVALCATx`, `BASECATx`)](#cat)
+-   [Derive Criterion Variables (`CRITy`, `CRITyFL`,
+    `CRITyFLN`)](#crit_vars)
+-   [Remaining ADVS Set-up](#advs_end)
+
+## Read in Data {#readdata}
+
+To start, all data frames needed for the creation of `ADVS` should be
+read into the environment. This will be a company specific process. Some
+of the data frames needed may be `VS` and `ADSL`.
+
+```{r, message=FALSE, warning=FALSE}
+library(admiral)
+library(admiralmetabolic)
+library(pharmaversesdtm)
+library(dplyr, warn.conflicts = FALSE)
+library(lubridate)
+library(stringr)
+
+data("dm_metabolic")
+data("vs_metabolic")
+
+dm <- convert_blanks_to_na(dm_metabolic)
+vs <- convert_blanks_to_na(vs_metabolic)
+```
+
+```{r, include=FALSE }
+data("admiral_adsl")
+
+dm <- dm %>%
+  left_join(admiral_adsl %>% select(USUBJID, TRTSDT, TRTEDT), by = "USUBJID")
+```
+
+The `DM` domain is used as the basis for `ADSL`:
+
+```{r eval=TRUE}
+adsl <- dm %>%
+  select(-DOMAIN)
+```
+
+```{r, eval=TRUE, echo=FALSE}
+dataset_vignette(
+  adsl,
+  display_vars = exprs(USUBJID, RFSTDTC, COUNTRY, AGE, SEX, RACE, ETHNIC, ARM, ACTARM)
+)
+```
+
+## Initial ADVS Set-up {#advs_start}
+
+The following steps are to merge ADSL variables with the source data and
+derive the usual ADVS analysis variables. The `{admiral}` [Creating a
+BDS Finding ADaM vignette](https://pharmaverse.github.io/admiral/articles/bds_finding.html)
+gives detailed guidance on all these steps.
+
+```{r, include=FALSE}
+adsl <- adsl %>%
+  mutate(
+    TRT01P = ARM,
+    TRT01A = ACTARM
+  )
+
+adsl_vars <- exprs(TRTSDT, TRTEDT, TRT01P, TRT01A)
+
+advs <- derive_vars_merged(
+  vs,
+  dataset_add = adsl,
+  new_vars = adsl_vars,
+  by_vars = exprs(STUDYID, USUBJID)
+)
+
+advs <- derive_vars_dt(advs, new_vars_prefix = "A", dtc = VSDTC)
+
+advs <-
+  derive_vars_dy(advs, reference_date = TRTSDT, source_vars = exprs(ADT))
+```
+
+The only difference here would be the the additional parameters
+collected in metabolic trials - `BMI`, `HIPCIR` and `WSTCIR`. Next step
+is to assign parameter level values such as `PARAMCD`, `PARAM`,
+`PARAMN`, `PARCAT1`, etc.,and for this a lookup can be created based on
+the SDTM `--TESTCD` value to join to the source data.
+
+```{r, include=FALSE}
+# `VSTESTCD` | `PARAMCD` | `PARAM` | `PARAMN` | `PARCAT1` | `PARCAT1N`
+# --------- | --------- | -------- | ------- | --------- | ----------
+# BMI | BMI | Body Mass Index (kg/m^2) | 1 | Subject Characteristic | 1
+# HIPCIR | HIPCIR | Hip Circumference (cm) | 3 | Subject Characteristic | 1
+# WSTCIR | WSTCIR | Waist Circumference (cm) | 5 | Subject Characteristic | 1
+```
+
+```{r, echo=TRUE, message=FALSE}
+param_lookup <- tribble(
```

Comment:
This looks good now - only ask is that you right-align the table - [see here](https://pharmaverse.github.io/admiral/dev/articles/visits_periods.html#creating-the-periodphase-reference-dataset-from-ex) for an example.

Reply:
@manciniedoardo Styler throws an error with right aligned lookup table. Have kept the original one for now.